### ISSUE TITLE
Regarding the code in liif.py

### ISSUE THREAD

**Issue by Swaraj-72** [yinboc/liif#34]
Hi, I have found it difficult to understand the purpose of rx, ry in the code below. Kindly let me know the same. Thank you.
<img width="115" alt="Issue" src="https://user-images.githubusercontent.com/65396498/125473328-35b509fe-fb80-41e9-9fb4-2ba4f7494fef.png">

**Comment by DongHwanJang** [comment]
Though I'm not the author, I think I can answer to your question partially.
rx, ry are used to find the nearest four points to the query point. If we find the nearest points for one pixel shifted point, it will be a closest point for the query point (which are used for the bilinear operation later).

But I have the same question about the flip operation and scale.
Don't we need to use
`rx = 2 / feat.shape[-2] /2` rather than `rx = 2 / feat.shape[-1] /2` since we are gonna filp the coordinate later?
It wouldn't be a problem for the square input, but might result in an unexpected output when we use rectangle input.
Please let me know if I have a missing point.
I asked a related question in here   #21

**Comment by Surayuth** [comment]
The coordinates of all centers are a constraint, by default, within [-1, 1]. This means the length of the side of each grid is [1 - (-1)]/N = 2/N where N is the number of grids in each direction. Thus, the shortest distance between the center of each grid to its border (radius, r) = (2/N)/2 = 2 / N / 2

**Comment by codeslake** [comment]
I believe flip is required for properly indexing x and y coordinates. 
The documentation for [`grid_sample()`](https://pytorch.org/docs/stable/generated/torch.nn.functional.grid_sample.html) states that,
> ...the size-2 vector grid[n, h, w] specifies input pixel locations x and y... .

As the [`coord`](https://github.com/yinboc/liif/blob/main/models/liif.py#L37) variable is basically a meshgride in (y, x) coordinate format ([here](https://github.com/yinboc/liif/blob/main/utils.py#L106-L114)), flip is required.

**Comment by axhiao** [comment]
For later readers, note that in grid_sample(),  the x-axis is along the width of an image and the y-axis is along its height. 

> I believe flip is required for properly indexing x and y coordinates. The documentation for [`grid_sample()`](https://pytorch.org/docs/stable/generated/torch.nn.functional.grid_sample.html) states that,
> 
> > ...the size-2 vector grid[n, h, w] specifies input pixel locations x and y... .
> 
> As the [`coord`](https://github.com/yinboc/liif/blob/main/models/liif.py#L37) variable is basically a meshgride in (y, x) coordinate format ([here](https://github.com/yinboc/liif/blob/main/utils.py#L106-L114)), flip is required.
